docs/pages/versions/unversioned/sdk/in-app-purchases.md

```diff
@@ -0,0 +1,317 @@
+
+#### Arguments
+
+- **callback (_(result: QueryResponse) => void_)** -- The callback function you want to run when there is an update to the purchases.
+
+#### Example
+```javascript
+// Set purchase listener
+setPurchaseListener(({ responseCode, results, errorCode }) => {
+    if (responseCode === ResponseCode.OK) {
+        for (const purchase of results) {
+            if (!purchase.acknowledged) {
+                console.log(`Successfully purchased ${purchase.productId}`);
+                // Process transaction here and unlock content
+
+                // Then when you're done...
+                finishTransactionAsync(purchase, true);
+            }
+        }
+    } else if (responseCode === ResponseCode.USER_CANCELED) {
+        console.log('User canceled');
+    } else { // Either received an error or deferred purchase (iOS only)
+        console.warn(
+            `Something went wrong with the purchase. Received response code ${responseCode} and errorCode ${errorCode}`
+        );
+    }
+});
+```
+
+### `InAppPurchases.getPurchaseHistoryAsync(refresh?: boolean)`
+
+Retrieves the user's previous purchase history.
+
+On Android, if refresh is set to `true` it will make a network request and return up to one entry per item even if that purchase is expired, canceled, or consumed. Use this if you want to sync purchases across devices or see purchases that are expired or consumed. If refresh is `false` it relies on the Play Store cache and returns the same result as `connectAsync`. An important caveat is that the return type when refresh is `true` is actually a subset of when it's `false`. This is because Android returns a `PurchaseHistoryRecord` which only contains the purchase time, purchase token, and product ID, rather than all of the attributes found in the `Purchase` type.
+
+On iOS, the refresh boolean is ignored and it returns the purchase history in the same way that `connectAsync` does. An important thing to note is that on iOS, Storekit actually creates a new transaction object every time you restore completed transactions, therefore the `purchaseTime` and `orderId` may be inaccurate if it's a restored purchase. If you need the original transaction's information you can use `originalPurchaseTime` and `originalOrderId`, but those will be 0 and an empty string respectively if it is the original transaction.
+
+#### Arguments
+
+- **refresh (_boolean_)** -- A boolean that indicates whether or not you want to make a network request to sync expired/consumed purchases and those on other devices (Android only)
+
+#### Returns
+
+A `Promise` that is resolved with a `QueryResponse` that contains an array of `Purchase` objects.
+
+#### Example
+```javascript
+const { responseCode, results } = await getPurchaseHistoryAsync();
+if (responseCode === ResponseCode.OK) {
+    results.forEach(result => {
+        // Handle purchase history
+    })
+}
+```
+
+### `InAppPurchases.purchaseItemAsync(productId: string, oldItem?: string)`
```

Comment:
if the function returns void, should it still be called `Async`? if it actually returns a promise, we should document when the promise is expected to resolve. (e.g. resolves as soon as the network request to the store completes successfully)

Reply:
It's async because it resolves a promise natively. It doesn't return anything since the response is handled by the callback in `setPurchaseListener`. Not sure what the convention is there but I thought every function that makes a native method call should end in `Async`.